hub/api/tests/test_api.py

```diff
@@ -211,6 +258,12 @@ def test_shape_property(memory_ds):
     assert fixed.shape.upper == (28, 28)
 
 
+@pytest.mark.xfail(raises=TensorMetaMismatchError, strict=True)
+def test_append_dtype_mismatch(memory_ds: Dataset):
```

Comment:
After I made the change to check the length and then call MisMatch error, this was failing. updated this to:
```
@pytest.mark.xfail(raises=TensorMetaMismatchError, strict=True)
def test_append_dtype_mismatch(memory_ds: Dataset):
    tensor = memory_ds.create_tensor("tensor", dtype="uint8")
    tensor.append(np.ones(100, dtype="float64"))
    tensor.append(np.ones(100, dtype="uint8"))
```

Reply:
uhmm... why though?

Reply:
if we are always overriding `dtype`, then what's the point of even having it???

Reply:
if anything, `dtype` for `generic` should be None, signifying elasticity (whatever the first sample is)

Reply:
but if they specify dtype, we should not just change it

Reply:
The above test will cause an exception as Dtype is set by the first example added to the dataset.
This is the same way, the shape is set after the first sample addition.

Reply:
i updated this, now when using default htype (`generic`) the dtype can be elastically set, otherwise errors are thrown